engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -1871,6 +1908,204 @@ public static long minus(@Nullable final ZonedDateTime dateTime1, @Nullable fina
+     */
+    public static Duration multiply(final long scalar, final Duration duration) {
+        return multiply(duration, scalar);
+    }
+
+    /**
+     * Multiply a duration by a scalar.
+     *
+     * @param duration the duration to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the duration multiplied by the scalar
+     */
+    public static Duration multiply(final Duration duration, final double scalar) {
+        if (duration == null || scalar == NULL_DOUBLE) {
+            return null;
+        }
+
+        if (Double.isNaN(scalar)) {
+            throw new DateTimeOverflowException("Scalar value is NaN");
+        }
+
+        final double product = duration.toNanos() * scalar;
+
+        if (product > Long.MAX_VALUE || product < Long.MIN_VALUE) {
+            throw new DateTimeOverflowException("Product value is too large to be cast to a long");
+        }
+
+        return Duration.ofNanos((long) product);
+    }
+
+    /**
+     * Multiply a duration by a scalar.
+     *
+     * @param duration the duration to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the duration multiplied by the scalar
+     */
+    public static Duration multiply(final double scalar, final Duration duration) {
+        return multiply(duration, scalar);
+    }
+
+    /**
+     * Multiply a period by a scalar.
+     *
+     * @param period the period to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the period multiplied by the scalar
+     */
+    public static Period multiply(final Period period, final int scalar) {
+        if (period == null || scalar == NULL_INT) {
+            return null;
+        }
+
+        return period.multipliedBy(scalar);
+    }
+
+    /**
+     * Multiply a period by a scalar.
+     *
+     * @param period the period to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the period multiplied by the scalar
+     */
+    public static Period multiply(final int scalar, final Period period) {
+        return multiply(period, scalar);
+    }
+
+    /**
+     * Multiply a period by a scalar.
+     *
+     * @param period the period to multiply
+     * @param scalar the scalar to multiply by
+     * @return {@code null} if either input is {@code null}; otherwise the period multiplied by the scalar
+     */
+    public static Period multiply(final Period period, final long scalar) {
+        if (period == null || scalar == NULL_LONG) {
+            return null;
+        }
+
+        if (scalar > Integer.MAX_VALUE || scalar < Integer.MIN_VALUE) {
+            throw new DateTimeOverflowException("Scalar value is too large to be cast to an int");
+        }
```

Comment:
Why provide a `long` version if this is the case?

Reply:
I don't like this argument. The same could be said for any int-based method "why doesn't it accept a long?". Should we create a version of this that works with `BigInteger`? If the answer is "no", I would say we shouldn't provide a `long` version. I would much prefer to address the root of the issue "I've got a logical int-column, but it's technically typed as long; how can I make it an int column?". In which case, you could just cast or do a check + cast.

Reply:
I'm with Devin, here. If we don't accept `longs` that are out of `int` range, it makes no sense to have a method that takes a `long`.

Reply:
I agree with @rbasralian on this.  Stuff like this happens frequently in real queries.

Reply:
If this is happening frequently in real queries, we should address the root cause of it: "Why are logical int columns typed as longs?".

Reply:
This isn't something we're doing elsewhere in our libraries. It isn't something users should expect from libraries generally.